src/main/java/com/uber/cadence/internal/sync/SyncDecisionContext.java

```diff
@@ -130,9 +154,18 @@ public WorkflowInterceptor getWorkflowInterceptor() {
   @Override
   public byte[] executeWorkflow(
       SyncWorkflowDefinition workflowDefinition, WorkflowExecuteInput input) {
-    return workflowDefinition.execute(input.getInput());
+    Span span = tracingPropagator.activateSpanForExecuteWorkflow(context);
+    try {
+      return workflowDefinition.execute(input.getInput());
+    } finally {
+      span.finish();
```

Comment:
also as before, it *might* be worth checking if there was an error (to the extent I understand what's happening)

Reply:
I'll add that in a later PR